apps/api/src/middlewares/html-template.middleware.ts

```diff
@@ -94,71 +78,89 @@ const locales = {
   }
 };
 
-const isFileRequest = (filename: string) => {
-  if (filename === '/assets/LICENSE') {
-    return true;
-  } else if (
-    filename.includes('auth/ey') ||
-    filename.includes(
-      'personal-finance-tools/open-source-alternative-to-de.fi'
-    ) ||
-    filename.includes(
-      'personal-finance-tools/open-source-alternative-to-markets.sh'
-    )
-  ) {
-    return false;
-  }
+@Injectable()
+export class HtmlTemplateMiddleware implements NestMiddleware {
+  private indexHtmlMap: { [languageCode: string]: string } = {};
 
-  return filename.split('.').pop() !== filename;
-};
+  constructor(private readonly i18nService: I18nService) {
+    try {
+      this.indexHtmlMap = SUPPORTED_LANGUAGE_CODES.reduce(
+        (map, languageCode) => ({
+          ...map,
+          [languageCode]: fs.readFileSync(
+            join(__dirname, '..', 'client', languageCode, 'index.html'),
+            'utf8'
+          )
+        }),
+        {}
+      );
+    } catch (error) {
+      console.error('Failed to load index.html files:', error);
```

Comment:
`Logger.error(...)`

`Failed to load index.html files` or `Failed to initialize index HTML map` ❓